Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSBacktrace.c

```diff
@@ -104,6 +106,41 @@ bool bsg_ksbt_isBacktraceTooLong(
     return true;
 }
 
+#if defined (__x86_64__) || defined(__arm64__)
+static bool isInMainImage(uintptr_t address) {
+    BSG_Mach_Header_Info *mainImage = bsg_mach_headers_get_main_image();
+    if (mainImage == NULL) {
+        return false;
+    }
+    return address >= (uintptr_t)mainImage->header &&
+        address < (uintptr_t)mainImage->header + mainImage->imageSize;
+}
+#endif
+
+static uintptr_t getFirstStackTraceAddress(const BSG_STRUCT_MCONTEXT_L *const machineContext) {
+    uintptr_t instructionAddress = bsg_ksmachinstructionAddress(machineContext);
+
+    if (!bsg_is_address_in_instruction_range(instructionAddress)) {
+        // These aren't the right addresses, but they do point to the beginning of
+        // the function that made the bad pointer call, which is better than nothing.
+        // We only do this if the address is in the main image, otherwise it could
+        // lead to a wild goose chase.
+#if defined (__x86_64__)
+        const uintptr_t tryAddress = machineContext->__ss.__r11;
+        if (isInMainImage(tryAddress)) {
```

Comment:
Maybe this should be `bsg_is_address_in_instruction_range` instead?

Many (most?!) apps these days are comprised of multiple modules / frameworks, so only considering the main image seems too restrictive.